common/model-views.cpp

```diff
@@ -3698,23 +3699,29 @@ namespace rs2
 
             //glTexParameterfv(GL_TEXTURE_2D, GL_TEXTURE_BORDER_COLOR, tex_border_color);
 
-            glBegin(GL_POINTS);
+            glBegin(GL_QUADS);
 
             auto vertices = last_points.get_vertices();
             auto tex_coords = last_points.get_texture_coordinates();
 
-            for (int i = 0; i < last_points.size(); i++)
-            {
-                if (vertices[i].z)
-                {
-                    glVertex3fv(vertices[i]);
-                    glTexCoord2fv(tex_coords[i]);
-                }
+#define THRESH 0.5f
+            auto width = vf_profile.width(), height = vf_profile.height();
+            for (int x = 0; x < width - 1; ++x) {
```

Comment:
The vertices are contained in a vector- not frame (2D-matrix).
In the vector all cells are guaranteed to have a valid depth. But each two successive member (i, i+1) are not guaranteed to correspond to pixels (i,i+1).
Let's discuss it

Reply:
@ev-mp this is not correct